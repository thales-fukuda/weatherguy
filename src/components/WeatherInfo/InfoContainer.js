import styled from 'styled-components';

const InfoContainer = styled.section`
  width: 100%;
  height: 85%;
  padding-top: 15%;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  opacity: ${props => (props.isWeatherDisplayed ? '1' : '0')};
  transition: 1s ease opacity;
  transition-delay: .5s;
  z-index: 1;

  @media (min-width: 720px) {
    padding-top: 5%;
  }
`;

export default InfoContainer;
