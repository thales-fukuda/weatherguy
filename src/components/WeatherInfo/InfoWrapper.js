import styled from 'styled-components';

const InfoWrapper = styled.section`
  width: 85%;
  display: flex;
  justify-content: space-around;
`;

export default InfoWrapper;
