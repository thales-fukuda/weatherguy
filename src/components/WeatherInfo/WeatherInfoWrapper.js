import { connect } from 'react-redux';
import WeatherInfo from './WeatherInfo';

const mapStateToProps = state => ({
  city: state.weather.city,
  weatherData: state.weather.weatherData,
  weatherStatus: state.weather.weatherStatus,
});

export default connect(
  mapStateToProps,
)(WeatherInfo);
