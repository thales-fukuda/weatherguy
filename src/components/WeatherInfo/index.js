import WeatherInfo from './WeatherInfoWrapper';

export default WeatherInfo;
