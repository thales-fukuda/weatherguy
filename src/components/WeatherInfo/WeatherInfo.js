import React from 'react';
import PropTypes from 'prop-types';

import InfoBox from '../blocks/InfoBox';
import InfoWrapper from './InfoWrapper';

const WeatherInfo = (props) => {
  const {
    weatherData,
  } = props;

  const weather = weatherData.main;

  const toCelsius = value => Math.floor(value - 273.15);
  const toFahrenheit = value => Math.floor(toCelsius(value) * 9 / 5 + 32);

  return (
    <>
      {
        weatherData.main
        && (
        <>
          <InfoWrapper>
            <InfoBox>
              <InfoBox.Text fontSize='62px'>{weatherData.name}</InfoBox.Text>
            </InfoBox>
          </InfoWrapper>
          <InfoWrapper>
            <InfoBox opacity='.5'>
              <InfoBox.Text fontSize='16px'>min</InfoBox.Text>
              <InfoBox.Text>{ `${toCelsius(weather.temp_min)} °C` }</InfoBox.Text>
              <InfoBox.Text>{ `${toFahrenheit(weather.temp_min)} °F` }</InfoBox.Text>
            </InfoBox>

            <InfoBox>
              <InfoBox.Text fontSize='62px'>{ `${toCelsius(weather.temp)} °C` }</InfoBox.Text>
              <InfoBox.Text fontSize='62px'>{ `${toFahrenheit(weather.temp)} °F` }</InfoBox.Text>
            </InfoBox>

            <InfoBox opacity='.5'>
              <InfoBox.Text fontSize='16px'>max</InfoBox.Text>
              <InfoBox.Text>{ `${toCelsius(weather.temp_max)} °C` }</InfoBox.Text>
              <InfoBox.Text>{ `${toFahrenheit(weather.temp_max)} °F` }</InfoBox.Text>
            </InfoBox>
          </InfoWrapper>

          <InfoWrapper>
            <InfoBox>
              <InfoBox.Icon className='wi wi-strong-wind' />
              <InfoBox.Text>{ `${weatherData.wind.speed} m/s` }</InfoBox.Text>
            </InfoBox>

            <InfoBox>
              <InfoBox.Icon className='wi wi-rain-wind' />
              <InfoBox.Text>{ `${weather.humidity} %` }</InfoBox.Text>
            </InfoBox>
          </InfoWrapper>
        </>
        )
      }
    </>
  );
};

WeatherInfo.propTypes = {
  weatherData: PropTypes.instanceOf(Object).isRequired,
};

export default WeatherInfo;
