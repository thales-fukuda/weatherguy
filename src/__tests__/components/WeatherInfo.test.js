/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import WeatherInfo from '../../components/WeatherInfo/WeatherInfo';

describe('<WeatherInfo/>', () => {
  describe('render()', () => {
    it('renders the component', () => {
      const fakeProps = {
        weatherData: {},
        weatherStatus: true,
      };
      const wrapper = shallow(<WeatherInfo {...fakeProps} />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
