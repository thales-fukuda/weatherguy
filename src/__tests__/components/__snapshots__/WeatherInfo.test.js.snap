// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherInfo/> render() renders the component 1`] = `
<styled.section
  isWeatherDisplayed={true}
/>
`;
